src/Plugin/views/field/TideNodeBulkForm.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+namespace Drupal\tide_site_restriction\Plugin\views\field;
+
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\node\Plugin\views\field\NodeBulkForm;
+
+/**
+ * {@inheritdoc}
+ */
+class TideNodeBulkForm extends NodeBulkForm {
+
+  /**
+   * Form constructor for the bulk form.
+   *
+   * @param array $form
+   *   An associative array containing the structure of the form.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The current state of the form.
+   */
+  public function viewsForm(&$form, FormStateInterface $form_state) {
```

Comment:
@vincent-gao looks like the method is handling both form rendering and form processing logic. It will be good to separate them for better standard. e.g - 
`protected function addBulkFormElements(&$form, FormStateInterface $form_state) {`
this part will handle the processing and then 

```
protected function addBulkFormActions(&$form) {
    if (!empty($form['actions'])) {
      // Ensure a consistent container for filters/operations in the view header.
      $form['header'] = [
        '#type' => 'container',
        '#weight' => -100,
      ];
      .......etc
```
to handle the for action.

Then call them in the viewsForm method - 

```
public function viewsForm(&$form, FormStateInterface $form_state) {
    // Make sure we do not accidentally cache this form.
    // @todo Evaluate this again in https://www.drupal.org/node/2503009.
    $form['#cache']['max-age'] = 0;

    // Add the tableselect javascript.
    $form['#attached']['library'][] = 'core/drupal.tableselect';

    $this->addBulkFormElements($form, $form_state);
    $this->addBulkFormActions($form);
  }
```